python_benchmarks/core.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import ctypes
+import torch
+from torch.profiler import profile, ProfilerActivity
+
+libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+for libname in libnames:
+    try:
+        cuda = ctypes.CDLL(libname)
+    except OSError:
+        continue
+    else:
+        break
+else:
+    raise OSError("could not load any of: " + " ".join(libnames))
+
+CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE = 38
+l2_cache_size = ctypes.c_int()
+device = torch.cuda.current_device()
+cuda.cuDeviceGetAttribute(
+    ctypes.byref(l2_cache_size), CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE, device
+)
+
+L2_CACHE_SIZE = l2_cache_size.value
+
+
+def clearL2Cache():
+    n_elements = L2_CACHE_SIZE // 4
+    x = torch.empty(n_elements, dtype=torch.float32, device="cuda", requires_grad=False)
+    y = torch.clone(x)
+
+
+class NVFBenchmark(object):
```

Comment:
I don't think it's necessary for this class to inherit `object`. Even if some of the users are still in Python 2, this class uses none of the features provided by `object` and inheriting from `object` makes it deviate from Python3's recommendation.

Reply:
> Even if some of the users are still in Python 2

Python 2 is already dead, and PyTorch does not support python 2 either.

Reply:
Thanks, Xiang -- it's great to know we don't need to support the legacy Python. Then, there are no reasons to inherit from `object`: https://stackoverflow.com/a/45062077.